packages/storage/src/indexed-db/index.ts

```diff
@@ -802,4 +784,12 @@ export class IndexedDb implements IndexedDbInterface {
       output: Value.fromJson(result.output),
     };
   }
+
+  async hasNativeAssetBalance(): Promise<boolean> {
+    const key = 'KeqcLzNx9qSH5+lcJHBB9KNW+YPrBk5dKzvPMiypahA=';
```

Comment:
same
you can get the `stakingAssetId` from the prax registry instead of a string constant

Reply:
@grod220 I modified the function to query the registry, but this is happening for every call. Do we instead want to create a global variable `STAKING_TOKEN_ASSET_ID` since the staking token asset id won't change?

Reply:
and if the registry becomes unaccessible for some reason, won't this block transaction creation?

Reply:
The registry is bundled within the NPM package. That means, it no longer makes a live http call. I just checks the existing bundle by chain Id. That means it should only fail if the chainId doesn't exist in the registry (which is not likely if they made it to this step). Also, it doesn't need to be async either, you can do:

```ts
  fetchStakingTokenId(): AssetId {
    const registryClient = new ChainRegistryClient();
    const registry = registryClient.get(this.chainId);
    const stakingTokenMetadata = registry.getMetadata(registry.stakingAssetId);
    return getAssetId(stakingTokenMetadata);
  }
```